### PR TITLE
[MERGE] block-inaccessible-environment into development

### PR DESCRIPTION
**Pull request by LotteBase**
https://github.com/XRBASE/Ravel-Unity-02/pull/331

- [ ] dev environments are blocked (except for refresh) when running in app mode (and the other way around).
- [ ] when building an environment in the wrong mode, the application shows a pop-up and cancels the build.

### CODE REVIEW

File: Ravel-Creators/Assets/RavelCreator/Scripts/Editor/CreatorTools/CreateEnvironmentWindow.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 using System.IO;
```
Comment (Vanclyf) [inline Ravel-Creators/Assets/RavelCreator/Scripts/Editor/CreatorTools/CreateEnvironmentWindow.cs]: unused lib

File: Ravel-Creators/Assets/RavelCreator/Scripts/Editor/CreatorTools/RavelToolbar.cs [hunk 2]
```diff
@@ -90,12 +90,12 @@ public class RavelToolbar
         
         //preview environment in ravel web
         if (GUILayout.Button(new GUIContent("Preview", "Build and preview asset bundle"), RavelEditorStying.txtBtnSmall)) {
```
Comment (Vanclyf) [inline Ravel-Creators/Assets/RavelCreator/Scripts/Editor/CreatorTools/RavelToolbar.cs]: RefreshConfig(), We could make the sceneconfiguration a singleton so we don't need to call FinndobjectsofType. Plus There is only one sceneconfiguration anyway per scene

File: Ravel-Creators/Assets/RavelCreator/Scripts/Editor/RavelEditor.cs [hunk 3]
```diff
@@ -245,7 +256,7 @@ public static class RavelEditor
         float m = mask.width / mask.height;
         float t = (float)tex.width / tex.height;
 
-        if (Mathf.Abs(m - t) <= MathBuddy.FloatingPoints.LABDA) {
+        if (Mathf.Abs(m - t) <= FloatingPoints.LABDA) {
             return coords;
         }
         if (m > t) {
```
Comment (Vanclyf) [inline Ravel-Creators/Assets/RavelCreator/Scripts/Editor/RavelEditor.cs]: Delete duplicate DrawTextureScaleHeightGUI()

File: Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Backend/WebRequests/RavelWebRequest.cs [hunk 4]
```diff
@@ -300,11 +312,22 @@ public class TokenWebRequest : RavelWebRequest {
...
+        private void AddToken() {
+            string token = GetToken();
+            
+            if (string.IsNullOrEmpty(token)) {
+                _error = InternalError.MissingToken;
+                return;
+            }
+            
+            AddHeader("Authorization", "Bearer " + token);
         }
```
Comment (Vanclyf) [inline Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Backend/WebRequests/RavelWebRequest.cs]: Delete unused duplicate  public RavelWebRequest(string url, WWWForm form, Dictionary<string, string> parameters = null, Dictionary<string, string> header = null)

File: Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Configuration/NetworkConfig.cs [hunk 5]
```diff
@@ -74,7 +74,7 @@ private string GetBaseUrl(AppMode mode, bool debugValue = false)
                     case AppMode.Test:
```
Comment (Vanclyf) [inline Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Configuration/NetworkConfig.cs]: line 13; REGION is unused

File: Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Configuration/NetworkConfig.cs [hunk 6]
```diff
@@ -181,20 +181,16 @@ public static NetworkConfig Load()
...
             return cfg;
         }
 
         public enum AppMode
         {
+            Unknown = -1,
             Development = 0,
             Test = 1,
-            Live = 2,
+            App = 2, //live
```
Comment (Vanclyf) [inline Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Configuration/NetworkConfig.cs]: Remove comment

**Review by Vanclyf — CHANGES_REQUESTED** [review]

File: Ravel-Creators/Assets/RavelCreator/Scripts/Editor/CreatorTools/RavelToolbar.cs (hunk 2 above)
Comment (LotteBase) [reply Ravel-Creators/Assets/RavelCreator/Scripts/Editor/CreatorTools/RavelToolbar.cs]: This is an editor script, so using a singleton does not work as the awake is not called. I did add a pop up that say's only one is allowed when you add a second one.

File: Ravel-Creators/Assets/RavelCreator/Scripts/Editor/RavelEditor.cs (hunk 3 above)
Comment (LotteBase) [reply Ravel-Creators/Assets/RavelCreator/Scripts/Editor/RavelEditor.cs]: This is not a duplicate. One method is `width` the other is `height`. 

It is not used (right now), but the height version scales the width of the image based on the heigth and the width version scales the height, based on the width.

I feel like this method may be needed at some later point in creating the editor codes.

File: Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Backend/WebRequests/RavelWebRequest.cs (hunk 4 above)
Comment (LotteBase) [reply Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Backend/WebRequests/RavelWebRequest.cs]: This webrequest is used in Ravel (non-creator). I've kept the webrequest files the same, so we can keep them in sync.

File: Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Configuration/NetworkConfig.cs (hunk 5 above)
Comment (LotteBase) [reply Ravel-Creators/Assets/RavelCreator/Scripts/Ravel/Configuration/NetworkConfig.cs]: Same as above, this is used in the non editor ravel.

**Review by Vanclyf — APPROVED** [review]

**Pull request merged**